statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -261,85 +280,263 @@ def var_loglike(resid, omega, nobs):
+    return VAR(Y, nlags).fit(trend=trend)
+
+class VAR(tsbase.TimeSeriesModel):
+    __doc__ = tsbase._tsa_doc % {"model" : _var_model_doc,
+                                 "params" : _var_params_doc,
+                                 "extra_params" : tsbase._missing_param_doc,
+                                 "extra_sections" : _var_reference_doc}
+    def __init__(self, Y, nlags=None, dates=None, names=None, freq=None,
+                 missing='none'):
+        if nlags is None:
+            from warnings import warn
+            warn("In the 0.6.0 release nlags will not be optional in the "
+                 "model constructor.", FutureWarning)
+        else:
+            self.k_ar = nlags
+        super(VAR, self).__init__(Y, None, dates, freq, missing=missing)
         if self.endog.ndim == 1:
-            raise ValueError("Only gave one variable to VAR")
-        self.y = self.endog #keep alias for now
-        self.neqs = self.endog.shape[1]
+            raise ValueError("Only one variable given to VAR. Expects > 1.")
+        if names is not None:
+            import warnings
+            warnings.warn("The names argument is deprecated and will be "
+                          "removed in 0.6.0. Use a pandas object or structured "
+                          "array for names.", FutureWarning)
+            self.names = names
+        else:
+            self.names = self.endog_names
+        self.Y = self.endog #keep alias for now
+        self.neqs = self.Y.shape[1]
+
 
     def _get_predict_start(self, start, k_ar):
         if start is None:
             start = k_ar
         return super(VAR, self)._get_predict_start(start)
 
+    def loglike(self, params, k_ar=None):
+        r"""
+        Returns the value of the VAR(p) log-likelihood.
+
+        Parameters
+        ----------
+        params : ndarray
+            The parameters in the order given in VARResults.
+        k_ar : int, optional
+            The number of lags if the equation has not yet been fit.
+
+        Returns
+        -------
+        llf : float
+            The value of the loglikelihood function for a VAR(p) model
+
+        Notes
+        -----
+        The loglikelihood function for the VAR(p) is
+
+        .. math::
+
+           -\left(\frac{nobs}{2}\right)
+           \left(\ln\left|cov_sigma_mle\right|-neqs\ln\left(2\pi\right)-neqs\right)
+        """
+        if k_ar is None:
+            try:
+                k_ar = self.k_ar
+            except:
+                raise ValueError("Must give k_ar, if the model has not been "
+                                 "fit")
+
+        Y_sample = self.Y[k_ar:]
+        nobs = len(Y_sample)
+        X = util.get_lagged_y(Y, k_ar, trend=trend)
+        resid = Y_sample - np.dot(X, params)
+        sse = np.dot(resid.T, resid)
+        cov_resid_mle = sse / nobs
+        return var_loglike(resid, cov_resid_mle, nobs)
+
     def predict(self, params, start=None, end=None, lags=1, trend='c'):
```

Comment:
same problem here: lags and trend should not be arguments to predict. They need to be used from the model.